src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -656,10 +656,51 @@ def update_period(self):
             exit_code, _, err = mgr.send_rgwadmin_command(rgw_update_period_cmd)
             if exit_code > 0:
                 raise DashboardException(e=err, msg='Unable to update period',
-                                         http_status_code=500, component='rgw')
+                                         http_status_code=400, component='rgw')
         except SubprocessError as error:
             raise DashboardException(error, http_status_code=500, component='rgw')
 
+    def edit_realm(self, realm_name: str, default: bool, new_realm_name: str):
+        rgw_realm_edit_cmd = []
+        cmd_edit_realm_options = []
+        if new_realm_name == realm_name:
+            if str_to_bool(default):
+                rgw_realm_edit_cmd = ['realm', 'default']
+                cmd_edit_realm_options = ['--rgw-realm', realm_name]
+                rgw_realm_edit_cmd += cmd_edit_realm_options
+                try:
+                    exit_code, _, err = mgr.send_rgwadmin_command(rgw_realm_edit_cmd)
+                    if exit_code > 0:
+                        raise DashboardException(e=err, msg='Unable to set {} as default realm'.format(realm_name),  # noqa E501  #pylint: disable=line-too-long
+                                                 http_status_code=500, component='rgw')
+                except SubprocessError as error:
+                    raise DashboardException(error, http_status_code=500, component='rgw')
+            else:
+                raise DashboardException(msg='The realm already exists',
+                                         http_status_code=400, component='rgw')
```

Comment:
Let's see, you can if `enabled == true` and if it is `false`, it means the realm already exists. There wasn't any realm checking performed, only a name comparison. 

Here I wouldn't throw an error because if you don't change anything I believe it should be idempotent.